rep-0147.rst

```diff
@@ -0,0 +1,365 @@
+Content-Type: text/x-rst
+Created: 16-May-2016
+Post-History: 30-Aug-2002
+
+
+Abstract
+========
+
+This REP is written to provide a common interface for aerial vehicles.
+This interface is designed to promote interoperability and reusability of core functionality between diverse aerial vehicles.
+By leveraging the interfaces defined in this REP we both tools and libraries can be designed to be reusable between different frameworks and vehicles.
+It is expected that aerial vehicles can and will implement additional interfaces in addition to these core interfaces.
+When choosing an interface these should be considered the default and only change away from them for a good reason.
+
+These interfaces are designed to be relevant for all types of aerial vehicles whether multi-copter, fixed-wing, hybrid, transitional or other.
+Like ground robots, the core abstraction is the same for all vehicles regardless of their method of motivation.
+
+Rationale
+=========
+
+There are a lot of developers around the world using ROS on aerial vehicles.
+As such there are also many different implementations which are similiar but not exactly compatible.
+A short list of existing known bridges and implementations follows: 
+
+ - astec_mav_framework [#astec]_
+ - autopilot_bridge [#autopilot_bridge]_
+ - CRATES [#crates]_
+ - dji_sdk_dji2ros [#djisdk]_
+ - hector_quadrotor [#hector]_
+ - mav_tools [#mav_tools]_
+ - mavlink2ros [#mavlink2ros]_
+ - mavros [#mavros]_
+ - roscopter [#roscopter]_
+ - rospilot [#rospilot]_
+
+Due to a lack of common interfaces the majority of the above packages cannot be used together.
+By implementing common interfaces elements of each package can be used in conjunction with elements of a different package.
+This will allow much better collaboration between projects and allow a reduction in duplicate efforts to implement the same functionality.
+
+There are also a lot of interfaces already well developed from other domains which can be brought into the aerial vehicle domain.
+This REP seeks to also integrate as many of those reusable interfaces as well to provide the greatest compatibilty with those other domains.
+
+Other interface defintions
+==========================
+
+There are a lot of messages already defined.
+These messages definitions were used to help define this set. 
+
+ - DroneKit [#dronekit]_
+ - mav_msgs [#mav_msgs]_
+ - mavlink [#mavlink]_
+ - mavlink2 [#mavlink2]_
+ - mavros_msgs [#mavros]_
+ - nav_msgs [#nav_msgs]_
+ - trajectory_msgs [#trajectory_msgs]_
+
+
+
+Reference Abstractions
+======================
+
+For this document we will use the following abstractions of the architecture.
+These abstractions are used to define the interfaces, however it is not required that a given system implement all the interfaces.
+A given system may implement subsets of these interfaces.
+
+
+Control Abstractions
+--------------------
+
+For controlling an aerial vehicle we will use the following terms for the interfaces.
```

Comment:
I like the abstract interface for ROS messages. Their purpose is to provide a common language across many different system implementations.

If there are widely-used vehicle-specific control interfaces, we should provide a package that translates between those interfaces and the standard ROS messages. If that is not possible, then I suppose there must a problem with the selected abstraction.